File: betterrandom/src/test/java/io/github/pr0methean/betterrandom/prng/adapter/ReseedingSplittableRandomAdapterDemo.java
package io.github.pr0methean.betterrandom.prng.adapter;

import io.github.pr0methean.betterrandom.seed.SecureRandomSeedGenerator;
import io.github.pr0methean.betterrandom.seed.SeedException;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.ScheduledThreadPoolExecutor;
import java.util.concurrent.TimeUnit;

public enum ReseedingSplittableRandomAdapterDemo {
  ;

  private static final String[] VALUE_LABELS =
      {"A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2"};
  private static final String[] SUIT_LABELS = {"♥️", "♣️", "♦️", "♠️"};

  public static void main(final String[] args) throws SeedException, InterruptedException {
    final String[] cards = new String[52];
    int i = 0;
    for (final String suit : SUIT_LABELS) {
      for (final String value : VALUE_LABELS) {
        cards[i] = value + suit;
        i++;
      }
    }
    final ThreadLocal<List<String>> deckCopies =
        ThreadLocal.withInitial(() -> Arrays.asList(cards.clone()));
    final ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(4);
    final ReseedingSplittableRandomAdapter random =
        new ReseedingSplittableRandomAdapter(SecureRandomSeedGenerator.DEFAULT_INSTANCE);
    for (i = 0; i < 20; i++) {
      executor.submit(() -> {
        final List<String> deck = deckCopies.get();
        Collections.shuffle(deck, random);
        System.out.format("North: %s%nEast: %s%nSouth: %s%nWest: %s%n%n",
            String.join(",", deck.subList(0, 13)), String.join(",", deck.subList(13, 26)),
            String.join(",", deck.subList(26, 39)), String.join(",", deck.subList(39, 52)));
      });
    }
    executor.shutdown();
    executor.awaitTermination(1, TimeUnit.MINUTES);
  }
}

File: betterrandom/src/test/java/io/github/pr0methean/betterrandom/prng/adapter/ReseedingSplittableRandomAdapterDemo.kt
package io.github.pr0methean.betterrandom.prng.adapter

import io.github.pr0methean.betterrandom.seed.DefaultSeedGenerator.DEFAULT_SEED_GENERATOR

import io.github.pr0methean.betterrandom.seed.SeedException
import java.util.Arrays
import java.util.Collections
import java.util.concurrent.ScheduledThreadPoolExecutor
import java.util.concurrent.TimeUnit

enum class ReseedingSplittableRandomAdapterDemo {
    ;

    companion object {

        private val VALUE_LABELS = arrayOf("A", "K", "Q", "J", "10", "9", "8", "7", "6", "5", "4", "3", "2")
        private val SUIT_LABELS = arrayOf("♥️", "♣️", "♦️", "♠️")

        @Throws(SeedException::class, InterruptedException::class)
        @JvmStatic
        fun main(args: Array<String>) {
            val cards = arrayOfNulls<String>(52)
            var i = 0
            for (suit in SUIT_LABELS) {
                for (value in VALUE_LABELS) {
                    cards[i] = value + suit
                    i++
                }
            }
            val deckCopies = ThreadLocal.withInitial { Arrays.asList<String>(*cards.clone()) }
            val executor = ScheduledThreadPoolExecutor(4)
            val random = ReseedingSplittableRandomAdapter.getInstance(DEFAULT_SEED_GENERATOR)
            i = 0
            while (i < 20) {
                executor.submit {
                    val deck = deckCopies.get()
                    Collections.shuffle(deck, random)
                    System.out.format("North: %s%nEast: %s%nSouth: %s%nWest: %s%n%n",
                            deck.subList(0, 13).joinToString(","), deck.subList(13, 26).joinToString(","),
                            deck.subList(26, 39).joinToString(","), deck.subList(39, 52).joinToString(","))
                }
                i++
            }
            executor.shutdown()
            executor.awaitTermination(1, TimeUnit.MINUTES)
        }
    }
}
